auth/auth.go

```diff
@@ -60,16 +61,22 @@ type Resource struct {
 
 // Account provided by an auth provider
 type Account struct {
-	// ID of the account (UUIDV4, email or username)
+	// ID of the account e.g. email
 	ID string `json:"id"`
-	// Secret used to renew the account
-	Secret string `json:"secret"`
+	// Type of the account, e.g. service
+	Type string `json:"type"`
+	// Provider who issued the account
+	Provider string `json:"provider"`
+	// RefreshToken used to renew the account
+	RefreshToken string `json:"refresh_token"`
```

Comment:
Rather than storing refresh token in the account we should probably store it with the token itself. As its an implementation detail and related to post token usage.

Reply:
If we do this then we'd need to return an initial token as part of generation, either as an additional return value or as an attribute of account.

Reply:
I think creating an account is not the same as logging into the account. So having the separate call to create the token is not an issue right?

Reply:
This would require us to do no verification in Token(), since the raw account was have nothing to verify itself. I think it's cleaner to have the refreshtoken in account.

Reply:
Theres no place I've ever seen that stores refresh token in an account because its a thing generated at the time of creating a new token. I don't think it makes sense in go-micro to store it as part of the Account struct. It does make sense to exist in the Token itself. And for the auth service to store the refresh token as part of the token separately to verify it.